### PR TITLE
8259226: jextract should differentiate between clang errors vs other runtime issues

### PR DESCRIPTION
**Pull request by sundararajana**
CLANG_ERROR error code added. Also OUTPUT_ERROR exit code is used as expected.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [ ] Change must be properly reviewed

### Issue
 * [JDK-8259226](https://bugs.openjdk.java.net/browse/JDK-8259226): jextract should differentiate between clang errors vs other runtime issues

### Download
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/429/head:pull/429`
`$ git checkout pull/429`

### CODE REVIEW

**Review by mcimadamore — COMMENTED** [review]
While the changes look good, I think that the category CLANG_ERROR seems too broad. I think it would be useful to distinguish between genuine errors (e.g. syntax error in header) and an error that occurs because of some issues in the internal parsing machinery. The latter to me seems more a case of RUNTIME_ERROR than CLANG_ERROR. I have pointed out some examples in the code.

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/AccessSpecifier.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public int value() {
     public final static AccessSpecifier valueOf(int value) {
         AccessSpecifier x = lookup.get(value);
         if (null == x) {
-            throw new NoSuchElementException("Invalid AccessSpecifier kind value: " + value);
+            throw new ClangException("Invalid AccessSpecifier kind value: " + value);
```
Comment (mcimadamore) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/AccessSpecifier.java:64]: Should this be a CLANG_ERROR?

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/CallingConvention.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ public int value() {
     public final static CallingConvention valueOf(int value) {
         CallingConvention x = lookup.get(value);
         if (null == x) {
-            throw new NoSuchElementException();
+            throw new ClangException("Invalid calling convention: " + value);
```
Comment (mcimadamore) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/CallingConvention.java:71]: Should this be a CLANG_ERROR?

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/CursorKind.java [hunk 3]
```diff
@@ -293,7 +293,7 @@ public int value() {
     public final static CursorKind valueOf(int value) {
         CursorKind x = lookup.get(value);
         if (null == x) {
-            throw new NoSuchElementException("Invalid Cursor kind value: " + value);
+            throw new ClangException("Invalid Cursor kind value: " + value);
```
Comment (mcimadamore) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/CursorKind.java:296]: Should this be a CLANG_ERROR?

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/EvalResult.java [hunk 4]
```diff
@@ -71,7 +71,7 @@ public long getAsInt() {
             case Integral:
                 return getAsInt0();
             default:
-                throw new IllegalStateException("Unexpected kind: " + kind);
+                throw new ClangException("Unexpected kind: " + kind);
```
Comment (mcimadamore) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/EvalResult.java:74]: Should this be a CLANG_ERROR? Same for others in this file

**Comment by sundararajana** [comment]
Closing this PR. Will recreate new one.

**Pull request closed without merging**